lib/context/selectors/index.js

```diff
@@ -83,7 +86,19 @@ const context = createSelectorTree({
     binary: createLeaf(
       [contexts],
 
-      (contexts) => contexts.indexForBinary
+      (contexts) => (
+        (binary) => {
+          // trim off possible constructor args, one word at a time
+          // HACK until there's better CREATE semantics
+          let index = undefined;
+          while (index === undefined && binary) {
```

Comment:
`binary.length` maybe?

Reply:
I like this loop.

Reply:
I checked, and empty strings are falsey. I chose this approach in case `binary` is given as null or something.

Reply:
Yes. Makes sense. I'm getting this in the chrome console:
```javascript
> "" === false
< false
> "" == false
< true
> "x" == true
< false
```

Reply:
I might follow Ben's choice to use Typescript and then you won't have to scold me for being a type-waffler

Reply:
Oh, but doesn't the above suggest that a string with length isn't inherently 'truthy'.

Reply:
It's truthy, but not true... `while (condition)` means the same thing as `while (!!condition)`; `("x" ? true : false) === true`, e.g.

Reply:
Ok fine. But if you don't start making mistakes I'm going to stop reviewing your code!!!!

Reply:
Actually I'm still not convinced. I don't see how "0x606060" can ever evaluate to true:
```javascript
> "0x606060" == true
< false
> "0x606060" === true
< false
```

Reply:
Sorry for my creeping nature, but:
```
if ("0x606060") console.log("hi")
hi
```

Reply:
Ok I give up.

Reply:
this looks like a cool resource: https://www.sitepoint.com/javascript-truthy-falsy/

but checkout https://javascriptweblog.wordpress.com/2011/02/07/truth-equality-and-javascript/, apparently in `"0x606060" == true`, the string and bool both get typed to numbers and the string ends up being `NaN`

Reply:
Thanks @seesemichaelj, awesome.

Reply:
I guess if you're ever trying to evaluate truthy status, use conditionals instead of comparisons to true/false because the ECMA standard can cause weird type conversions when using comparisons